packages/runtime/src/utils/setAttribute.ts

```diff
@@ -18,9 +18,16 @@ export function setAttribute(
       }
       break
     case 'value':
-    case 'type':
-      ;(elem as any)[attr] = toBoolean(value) ? String(value) : undefined
+    case 'type': {
+      let val = value
+      if (elem instanceof HTMLProgressElement) {
```

Comment:
Maybe a super short comment here for the future, or link to the original issue

Reply:
Done